homeassistant/components/envisalink/__init__.py

```diff
@@ -118,7 +119,7 @@ async def async_setup(hass, config):
     zones = conf.get(CONF_ZONES)
     partitions = conf.get(CONF_PARTITIONS)
     connection_timeout = conf.get(CONF_TIMEOUT)
-    sync_connect = asyncio.Future()
+    sync_connect: asyncio.Future = asyncio.Future()
```

Comment:
I don't know why mypy is unable to deduce the hint here, but I had to add it manually for it to pass.

Reply:
It doesn't tell what the future entails.

```suggestion
    sync_connect: asyncio.Future[bool] = asyncio.Future()
```

Reply:
Ah that makes sense... similar to `dict` and `list`.
This is adjusted now.

Reply:
`Future[bool]` is ok in Python 3.9, but not in Python 3.8
I've had to make further changes but I think I've nailed it now.

Reply:
Well, you are in luck as we will be dropping Python 3.8 today 😉

Reply:
Dropped :)